crates/lib/src/cli.rs

```diff
@@ -961,6 +1032,22 @@ async fn rollback(opts: RollbackOpts) -> Result<()> {
     let sysroot = &get_storage().await?;
     crate::deploy::rollback(sysroot).await?;
 
+    if opts.queue_soft_reboot {
+        // Get status before rollback to check soft-reboot capability
+        let host = crate::status::get_status_require_booted(sysroot)?.2;
```

Comment:
Right so this is a bit tricky; the `OstreeSysroot` object has "interior mutability" even though we only have a `&` and not a `&mut` and so when we call APIs like `write_deployments` it ends up changing some of its caches and reloads them.

But the `Host` is a bootc side thing in Rust and so follows Rust defaults and we only have a `&` (it also doesn't monitor the `ostree::Sysroot`) so we do need to re-fetch it after making changes.

A followup cleanup here is probably for the APIs that do mutations to return a new `Host by default.

But anyways this looks sane to me as is

Reply:
Actually the inverse of this is important: We need to audit we're not using a stale `Host` object after a mutation in the upgrade/switch cases

Reply:
Yes, this is confusing. Basically ostree maintains an ordered list of deployments at the lowest level. The concept of "rollback" is a more invented/higher level one determined by looking at that ordering. (Also it takes stateroots into account)